src/add-to-project.ts

```diff
@@ -113,6 +140,52 @@ export async function addToProject(): Promise<void> {
   core.debug(`Project node ID: ${projectId}`)
   core.debug(`Content ID: ${contentId}`)
 
+  // Next, use the GraphQL API to request the project's fields using the node ID.
+  const fieldsResp = await octokit.graphql<ProjectFieldsResponse>(
+    `query getProjectFields($projectId: ID!) {
+      node(id: $projectId) {
+        ... on ProjectV2 {
+          fields(first: 50) {
+            nodes {
+              ... on ProjectV2Field {
+                id
+                name
+              }
+              ... on ProjectV2IterationField {
+                id
+                name
+                dataType
+                configuration {
+                  iterations {
+                    startDate
+                    id
+                  }
+                }
+              }
+              ... on ProjectV2SingleSelectField {
+                id
+                name
+                dataType
+                options {
+                  id
+                  name
+                }
+              }
+            }
+          }
+        }
+      }
+    }`,
+    {
+      projectId,
+    },
+  )
+
+  const projectFields = fieldsResp.node?.fields.nodes ?? []
+
+  const results = await check.checkDictionaryInArray(fieldsObj, projectFields)
```

Comment:
This should be moved later on and should only be used if there's at least one field in input - if there are not fields, there's no need to make this call.